### PR TITLE
feature:  support to enable HTTPS for admin API

### LINKED ISSUE #1647: feature: support to enable HTTPS for admin API
Allow users to access the Admin API via HTTPS.
We can add a single flag to support this feature.

```
apisix:
  allow_admin:                  # http://nginx.org/en/docs/http/ngx_http_access_module.html#allow
    - 127.0.0.0/24              # If we don't set any IP list, then any IP access is allowed by default.
    - "::/64"
  port_admin: 9180              # use a separate port for Admin API
  https_admin: true             # enable HTTPS when use a separate port for Admin API.
                                # Admin API will use conf/apisix_admin_api.crt and conf/apisix_admin_api.key as certificate.
```

### PR DESCRIPTION
**Pull request by dabue**
### Summary

#1647 
support to enable HTTPS for admin API 
1. add https_admin in config.yaml
2. update bin/apisix
3. add crt files
4. update apisix_cli_test.sh
Fix #1647

### CODE REVIEW

**Comment by dabue** [comment]
> please add test cases for this feature

I will add the test cases.

File: bin/apisix [hunk 1]
```diff
@@ -286,7 +286,18 @@ http {
 
     {% if enable_admin and port_admin then %}
     server {
+        {%if https_admin then%}
```
Comment (membphis) [inline bin/apisix:289]: I think we need to update the `conf/config.yaml` file, add new field `https_admin`

**Comment by dabue** [comment]
I have updated it and added the test cases.



88786912@qq.com
 
From: YuanSheng Wang
Date: 2020-06-05 21:30
To: apache/incubator-apisix
CC: dabue; Author
Subject: Re: [apache/incubator-apisix] [feature] support to enable HTTPS for admin API (#1648)
@membphis commented on this pull request.


In bin/apisix:
> @@ -286,7 +286,18 @@ http {
 
     {% if enable_admin and port_admin then %}
     server {
+        {%if https_admin then%}

I think we need to update the conf/config.yaml file, add new field https_admin

File: conf/config.yaml [hunk 2]
```diff
@@ -54,6 +54,7 @@ apisix:
     - 127.0.0.0/24              # If we don't set any IP list, then any IP access is allowed by default.
   #   - "::/64"
   # port_admin: 9180              # use a separate port
+  # https_admin: true             # enable https
```
Comment (membphis) [inline conf/config.yaml:57]: need more comments:

```
                                # enable HTTPS when use a separate port for Admin API.
                                # Admin API will use conf/apisix_admin_api.crt and conf/apisix_admin_api.key as certificate.
```

File: .travis/apisix_cli_test.sh [hunk 3]
```diff
@@ -72,3 +72,47 @@ done
 
 sed -i '/dns_resolver:/,+4s/^#//'  conf/config.yaml
 echo "passed: system nameserver imported"
+
```
Comment (moonming) [inline .travis/apisix_cli_test.sh:75]: I think we should add test cases in `t/admin/`.
like https://github.com/apache/incubator-apisix/blob/master/t/admin/stream-routes-disable.t#L37 to change `config.yaml`, and call https request like https://github.com/apache/incubator-apisix/blob/master/t/plugin/redirect.t#L608

Comment (membphis) [reply .travis/apisix_cli_test.sh:75]: I think the current implementation is better.

The `conf/nginx.conf` is generated by `conf/config.yaml`, so we use `CLI` to test this feature is better.

The `conf/nginx.conf` in `Test:Nginx` was generated by `t/APISIX.pm`, it is very different for a real user.

Comment (membphis) [reply .travis/apisix_cli_test.sh:75]: @moonming wait for your response

Comment (moonming) [reply .travis/apisix_cli_test.sh:75]: I think the test coverage of add test case in shell is too low, e.g. I don't know what happens if I open https but send an http request, and it not test that the host is incorrect, it's basically the same as not testing at all.
Maybe we can merge this PR first, but test cases must added later.

**Comment by dabue** [comment]
The test cases only checked whether the added field  https_admin in config.yaml is in effect or not.

**Pull request merged**